### PR TITLE
Fix a couple bugs with the workshop survey dashboard

### PR DESCRIPTION
**Pull request by mehalshah**
1) Keys were not unique in facilitator view
2) If I was both a workshop organizer AND a facilitator, the facilitator view would show workshops I organized, not facilitated.

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx [hunk 1]
```diff
@@ -199,9 +199,9 @@ const SurveyResultsHeader = React.createClass({
           <td style={row['heading'] && styles.questionGroupCell}>{allWorkshopsForCourse[row['key']]}</td>
           <td style={row['heading'] && styles.questionGroupCell}>{allMyWorkshopsForCourse[row['key']]}</td>
           {
-            headings.slice(3).map((heading, i) => {
+            headings.slice(3).map((heading, j) => {
               return (
-                <td key={i} style={row['heading'] && styles.questionGroupCell}>
+                <td key={j} style={row['heading'] && styles.questionGroupCell}>
```
Comment (aoby) [inline apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx:204]: What is different here? It looks like you just renamed `i` to `j`.

Comment (mehalshah) [reply apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx:204]: Yeah - it was part of an inner loop where the outer loop variable was i

File: dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb [hunk 2]
```diff
@@ -33,6 +33,13 @@ class Api::V1::Pd::WorkshopsControllerTest < ::ActionController::TestCase
     assert_equal 1, JSON.parse(@response.body).length
   end
 
+  test 'with the facilitated param, workshop organizers only view workshops they facilitated' do
+    sign_in @organizer
+    get :index, params: {workshops_ive_facilitated: true}
+    assert_response :success
+    assert_equal 0, JSON.parse(@response.body).length
```
Comment (aoby) [inline dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb]: I'd like to see this test assert that a workshop I facilitated was returned, not just that the workshop I organizer was not returned.

Comment (mehalshah) [reply dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb]: Yup will do

File: dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb [hunk 3]
```diff
@@ -34,10 +34,14 @@ class Api::V1::Pd::WorkshopsControllerTest < ::ActionController::TestCase
   end
 
   test 'with the facilitated param, workshop organizers only view workshops they facilitated' do
+    workshop_2 = create(:pd_workshop, organizer: @organizer, facilitators: [@organizer])
```
Comment (aoby) [inline dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb:37]: This is organized and facilitated by me. Rename it to something like `workshop_i_facilitated` and remove the `organizer` argument (which will cause it to generate a new organizer)

After that, 🚢

**Pull request merged**